project/eScavenge/views.py

```diff
@@ -70,3 +71,64 @@ def teamPage(request):
     teamhistory = userpage.history.all()
     context = {'team': userpage, 'teamlist': teamlist, 'teamhistory': teamhistory, 'command': command}
     return render(request, 'teamPage.html', context)
+
+def gamemakerPage(request):
+    username = request.session.get('username')
+    message = ''
+    cli = CLI(COMMANDS)
+    if username != GM.username:
+        return HttpResponseForbidden()
+    if (request.method == 'POST'):
+        if (request.POST.get("selectGame")):
+            request.session['curgame'] = request.POST["gameObject"]
+    if('curgame' in request.session.keys()):
+        curgame = request.session.get('curgame')
+        try:
+            gameteam = Game.objects.get(name = curgame).teams.all()
+        except:
+            gameteam = []
+            message = "No Teams Yet!"
+        try: 
+            gamemarks = Game.objects.get(name = curgame).landmarks.all()
+        except:
+            gamemarks = []
+            message = message + " No Landmarks Yet!"
+        gamestatus = "NOT WORKING YET"
+        if (Game.objects.get(name = curgame).started is False and Game.objects.get(name = curgame).ended is False):
+            gamestatus = "Waiting To Start"
+        elif (Game.objects.get(name = curgame).started is True and Game.objects.get(name = curgame).ended is True):
+            gamestatus = "Game Ended"
+        elif(Game.objects.get(name = curgame).started is True and Game.objects.get(name = curgame).ended is False):
+            gamestatus = "Game is Running"
+        else:
+           gamestatus = "Uknown Error"
+    else:
+        gamestatus = "No Game Selected"
+        curgame = False
+        gameteam = False
+        gamemarks = False
+    games = Game.objects.all()
+    markcount = len(gamemarks)
```

Comment:
If this is the very first time the gamemaker has logged in, or the session has been cleared, game marks will be set to `False` which you can't call `len()` on. This crashes.